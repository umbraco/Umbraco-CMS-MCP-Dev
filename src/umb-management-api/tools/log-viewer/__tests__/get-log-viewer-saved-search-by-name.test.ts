import { PagedSavedLogSearchResponseModel } from "@/umb-management-api/schemas/index.js";
import GetLogViewerSavedSearchByNameTool from "../get/get-log-viewer-saved-search-by-name.js";
import { LogViewerSavedSearchBuilder } from "./helpers/log-viewer-saved-search-builder.js";
import { LogViewerTestHelper } from "./helpers/log-viewer-test-helper.js";
import { jest } from "@jest/globals";
import { z } from "zod";

const TEST_SEARCH_NAME = "_Test Saved Search";

describe("get-log-viewer-saved-search-by-name", () => {
  let originalConsoleError: typeof console.error;

  beforeEach(() => {
    originalConsoleError = console.error;
    console.error = jest.fn();
  });

  afterEach(async () => {
    // Clean up any test saved searches
    await LogViewerTestHelper.cleanup(TEST_SEARCH_NAME);
    console.error = originalConsoleError;
  });

  it("should get a saved search by name", async () => {
    // Create test saved search
    const builder = await new LogViewerSavedSearchBuilder()
      .withName(TEST_SEARCH_NAME)
      .withQuery("level:Error")
      .create();

    const result = await GetLogViewerSavedSearchByNameTool().handler(
      { name: TEST_SEARCH_NAME },
      { signal: new AbortController().signal }
    );

    // Verify the response using snapshot
    expect(result).toMatchSnapshot();

    // Parse the response and verify structure
    const response = JSON.parse(result.content[0].text as string);
    expect(response).toBeDefined();
    expect(response.name).toBe(TEST_SEARCH_NAME);
    expect(response.query).toBe("level:Error");
  });

  it("should handle non-existent saved search", async () => {
    const result = await GetLogViewerSavedSearchByNameTool().handler(
      { name: "NonExistentSearch" },
      { signal: new AbortController().signal }
    );

    // Verify the error response using snapshot
    expect(result).toMatchSnapshot();
  });

  it("should handle empty name parameter", async () => {
    const result = await GetLogViewerSavedSearchByNameTool().handler(
      { name: "" },
      { signal: new AbortController().signal }
    );

    const response = JSON.parse(result.content[0].text as string);
    LogViewerTestHelper.verifySavedSearchResponse(response);
  });
});
