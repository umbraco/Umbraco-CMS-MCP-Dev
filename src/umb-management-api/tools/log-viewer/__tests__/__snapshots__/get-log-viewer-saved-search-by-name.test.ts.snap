// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get-log-viewer-saved-search-by-name should get a saved search by name 1`] = `
{
  "content": [
    {
      "text": "{"name":"_Test Saved Search","query":"level:Error"}",
      "type": "text",
    },
  ],
}
`;

exports[`get-log-viewer-saved-search-by-name should handle non-existent saved search 1`] = `
{
  "content": [
    {
      "text": "Error using get-log-viewer-saved-search-by-name:
{
  "message": "Request failed with status code 404",
  "response": {
    "type": "Error",
    "title": "The saved search could not be found",
    "status": 404
  }
}",
      "type": "text",
    },
  ],
}
`;
